### DISCUSSION TITLE
Getting 'transport: authentication handshake failed: x509: certificate signed by unknown authority' error on Manager

### DISCUSSION THREAD

**Discussion by ihuxn4hi** [Security-Onion-Solutions/securityonion#9731] [Unsupported Versions]
Hi there,

I'm getting 'transport: authentication handshake failed: x509: certificate signed by unknown authority' errors on the Manager and not quite sure how to proceed.  

I've got a valid cert and key issued by Let's Encrypt located in /etc/pki/managerssl.crt and /etc/pki/managerssl.key, respectively.  In this post (https://github.com/Security-Onion-Solutions/securityonion/discussions/7759), @defensivedepth suggests to run the following:

`salt-call state.apply  ssl,nginx`

And also:

`so-nginx-restart --force`

But neither of those resolved the issue.  Does anyone have any suggestions?

Thanks!

**Comment by InfosecGoon** [comment]
Where are you seeing that error? Is it in a log file or are you seeing it in the browser when connecting to the Manager?

**Reply by ihuxn4hi** [reply]
Hi there and thank you very much for the response.  

I'm seeing it in the console as the detail of 'Unknown problem somewhere in the system' alerts, but it's in /var/log/syslog of the Manager (Ubuntu 20.04). 

Here's an example of the full log:

`Feb  7 19:15:34 so-manager launcher[5810]: {"caller":"request_queries.go:162","err":"rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: authentication handshake failed: x509: certificate signed by unknown authority\"","method":"RequestQueries","reauth":false,"res":"null","severity":"info","took":"59.441µs","ts":"2023-02-07T19:15:34.186896698Z","uuid":"7308bea5-66e8-4fde-a84f-7a50be1bb480"}`

It's the highest volume log in the console:

<img width="421" alt="image" src="https://user-images.githubusercontent.com/123765379/217343890-d7a5f0f3-3187-4565-a7c1-c949f5b645f2.png">

Thanks again!

**Reply by InfosecGoon** [reply]
This error is coming from Launcher, which is the wrapper script around the OSQuery agent - have you changed the IP of this Manager or otherwise modified its network settings since installation? Is this a standalone node or part of a distributed installation?

**Reply by ihuxn4hi** [reply]
Thank you again for the reply.  The IP and hostname are the same but when I generated the Let's Encrypt cert, it had be the FQDN so it would pass the DNS validation check. This is in a distributed deployment.

I'd read to update url_base and fleet_hostname in /opt/so/saltstack/local/pillar/global.sls and then run 'salt-call state.apply fleet.event_gen-packages' which I did so those are both the FQDN now.  Are there additional steps required that I missed?

Thank you again -- I really appreciate your help!

**Reply by ihuxn4hi** [reply]
Any thoughts on this?

Thank you very much!

**Reply by InfosecGoon** [reply]
Have you reinstalled the OSQuery package on the Security Onion node after generating the new one with the FQDN? That may be where the mismatch is.

**Reply by ihuxn4hi** [reply]
I'm so sorry it's taken so long to get back to you.  Is osquery a docker image?  I don't see that or any system package for osquery.  I have seven nodes, all of which are running 2.3.220.

```
# docker ps |awk {'print $NF'} |sort -u
NAMES
so-aptcacherng
so-dockerregistry
so-elastalert
so-elasticsearch
so-filebeat
so-fleet
so-grafana
so-idstools
so-influxdb
so-kibana
so-kratos
so-logstash
so-mysql
so-nginx
so-playbook
so-redis
so-sensoroni
so-soc
so-soctopus
so-telegraf
so-wazuh
```

When I download and try to install the 'deb-launcher.deb' package from the Manager, I get this:

```
# apt-get install ./deb-launcher.deb 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Note, selecting 'launcher-final' instead of './deb-launcher.deb'
launcher-final is already the newest version (5.1.1).
```
Thank you again -- I really appreciate your help!

**Reply by InfosecGoon** [reply]
Try "sudo dpkg -i ./deb-launcher.deb" instead.

**Reply by ihuxn4hi** [reply]
Thank you again for the reply.  I did that and re-deployed the deb-launcher.deb package but am getting the same SSL errors as soon as the service starts:

`Mar 10 17:39:34 so-test launcher[114965]: {"caller":"request_queries.go:162","err":"rpc error: code = Unavailable desc = connection error: desc = \"transport: authentication handshake failed: x509: certificate signed by unknown authority\"","method":"RequestQueries","reauth":false,"res":"null","severity":"info","took":"78.647µs","ts":"2023-03-10T17:39:34.285143184Z","uuid":"8c1613b9-afcc-4651-a313-6ff0cedf7557"}`

What do I need to do to get the correct cert included when the packages get rebuilt?

Thanks again!

**Reply by ihuxn4hi** [reply]
Any idea on this one?  I'd like to roll this out but can't while this error persists. 

Thank you very much!

**Reply by InfosecGoon** [reply]
Can you share your global pillar? From the Manager, "sudo salt-call pillar.get global".

**Reply by ihuxn4hi** [reply]
You got it -- here you go:

```
# salt-call pillar.get global
local:
    ----------
    airgap:
        False
    dockernet:
        172.17.0.0
    fleet_custom_hostname:
        None
    fleet_hostname:
        so-manager.seceng.mydomain.com
    fleet_ip:
        10.0.210.159
    fleet_manager:
        True
    fleet_node:
        False
    fleet_packages-timestamp:
        2023-03-01-20:30
    fleet_packages-version:
        5
    hnmanager:
        10.0.0.0/8,192.168.0.0/16,172.16.0.0/12
    ids:
        Suricata
    imagerepo:
        security-onion-solutions
    managerip:
        10.0.210.159
    mdengine:
        ZEEK
    pipeline:
        redis
    sensoronikey:
        weSsnQnA7D11YaMuHBKL
    soversion:
        2.3.220
    url_base:
        so-manager.seceng.mydomain.com
    wazuh:
        1
```

Thank you very much!

**Reply by ihuxn4hi** [reply]
Hi there!  Just wanted to check in on this.  I'm fine using the self signed cert on the Manager if necessary if it resolves the mismatch with the clients.  I appreciate your guidance!

**Reply by ihuxn4hi** [reply]
Sorry, I don't want to be a pain but I'd really love to get this resolved.

Thank you again for all of your help!
